lib/src/runner/browser/browser.dart

```diff
@@ -126,6 +126,8 @@ abstract class Browser {
     });
 
     // Swallow exceptions. The user should explicitly use [onExit] for these.
-    return onExit.catchError((_) {});
+    // However, surface application exceptions which are thrown due to startup
+    // issues with the browser.
+    return onExit.catchError((_) {}, test: (e) => e is! ApplicationException);
```

Comment:
I don't think this is the right place to handle this error. As the comment indicates, code that uses `Browser` is expected to handle errors itself through the `onExit` getter. That suggests that it's that code's responsibility to add special handling for application exceptions.

I believe the right place to address this is in `BrowserManager.start()`.